### PR TITLE
Get account and other information from tso profile, if present (re-implement).

### PR DESCRIPTION
**Pull request by fswarbrick**
Signed-off-by: Frank Swarbrick <fswarbrick@gmail.com>

## Proposed changes

Currently only the account number is pulled from the tso profile, if there is one. Since there is (currently) no interface asking the user to enter other parameters (regionSize et al), there is no way for said parameter(s) to be passed as part of the TSO command request.

This change will pull all of the TSO profile parameters, if they exist, and pass them as part of the TSO command request.

## Release Notes

Milestone:

Changelog:

Deprecated issueTsoCommand in ZoweExplorerApi/ZosmfCommandApi.
Replaced with issueTsoCommandWithParms.  New method takes IStartTsoParms as input in place of just acctNum string.
Updated TsoCommandHandler to use issueTsoCommandWithParms.
Updated TsoCommandHandler.unit.test to use issueTsoCommandWithParms.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [ ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

This is replacing draft PR https://github.com/zowe/vscode-extension-for-zowe/pull/1359.  My git skills were lacking and I couldn't update it successfully with my new changes.

### CODE REVIEW

**Comment by fswarbrick** [comment]
All unit tests have passed.
Not all integration tests are passing for me.  But they didn't pass for me even without changes, so I imagine it's (hopefully) just my environment.
@JillieBeanSim, can you explain why there is a need to check for the existence of my new method that replaces existing method issueTsoCommand?  Since both the API and the extension itself are within this same repo, won't the new method always be present?  I understand how another extension that uses the API might depend on the now deprecated method, but I don't see that being the case here.

**Comment by JillieBeanSim** [comment]
> @JillieBeanSim, can you explain why there is a need to check for the existence of my new method that replaces existing method issueTsoCommand? Since both the API and the extension itself are within this same repo, won't the new method always be present? I understand how another extension that uses the API might depend on the now deprecated method, but I don't see that being the case here.

The new method may not exist for extenders, the check in the method will first check if the new api call is present in the extender's api file.  If it is it will be used with the new parameters, and if not it will fall back to the old deprecated api call.  This is required to not break the functionality with extenders and give them time to change over to the new api call before the deprecated api call is removed, possibly in v2.0.

File: packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts [hunk 1]
```diff
@@ -435,13 +435,24 @@ export namespace ZoweExplorerApi {
...
 
+        /**
+         * Issues a TSO Command and returns a TsoSend API response.
+         *
+         * @param {string} command
+         * @param {zowe.IStartTsoParms} parms
+         * @returns {zowe.IIssueResponse>}
+         * @memberof ICommand
+         */
+        issueTsoCommandWithParms?(command: string, parms: zowe.IStartTsoParms): Promise<zowe.IIssueResponse>;
```
Comment (JillieBeanSim) [inline packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts:454]: parms should include the `?`, similar to the old API method for extenders that don't require this parameter.
`parms?: zowe.IStartTsoParms`

File: packages/zowe-explorer/src/command/TsoCommandHandler.ts [hunk 2]
```diff
@@ -230,10 +231,11 @@ export class TsoCommandHandler extends ZoweCommandProvider {
     /**
      * Allow the user to submit an TSO command to the selected server. Response is written
      * to the output channel.
-     * @param session The Session object
      * @param command the command string
+     * @param profile profile to be used
+     * @param tsoParams parameters (from TSO profile, when used)
      */
-    private async issueCommand(command: string, profile: imperative.IProfileLoaded, acctNum?: string) {
+    private async issueCommand(command: string, profile: imperative.IProfileLoaded, tsoParams: IStartTsoParms) {
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/command/TsoCommandHandler.ts:238]: same here with the ? with the parameter
`tsoParams?: IStartTsoParms`

**Review by JillieBeanSim — COMMENTED** [review]
@fswarbrick I am seeing an issue with an extender that only has the old API method, see image.  I am looking to see how that can be fixed.  Also, with the updates from master branch only 1 integration test is failing now (same as master) =)
image

File: packages/zowe-explorer/src/command/TsoCommandHandler.ts [hunk 3]
```diff
@@ -247,7 +249,17 @@ export class TsoCommandHandler extends ZoweCommandProvider {
                         title: localize("issueTsoCommand.command.submitted", "TSO command submitted."),
                     },
                     () => {
-                        return ZoweExplorerApiRegister.getCommandApi(profile).issueTsoCommand(command, acctNum);
+                        if (ZoweExplorerApiRegister.getCommandApi(profile).issueTsoCommandWithParms) {
+                            return ZoweExplorerApiRegister.getCommandApi(profile).issueTsoCommandWithParms(
+                                command,
+                                tsoParams
+                            );
+                        } else {
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/command/TsoCommandHandler.ts:257]: I was able to get it working for the extender with the old API by adding this check on tsoParams here before the call to the old API
```
let acctNum: string;
if (!tsoParams) {
    acctNum = undefined;
} else {
    acctNum = tsoParams.account;
}
```

Comment (JillieBeanSim) [reply packages/zowe-explorer/src/command/TsoCommandHandler.ts:257]: followed by 
```
return ZoweExplorerApiRegister.getCommandApi(profile).issueTsoCommand(
                                command,
                                acctNum
                            );
```

**Review by JillieBeanSim — APPROVED** [review]
LGTM! Thanks @fswarbrick for all of your work on this! All expected tests and checks are passing as well.

**Review by lauren-li — COMMENTED** [review]
This PR looks good to me so far! All unit tests are passing, and the functionality seems to work well in manual testing without regressions to extenders.

Unfortunately, my integration tests are acting up and crashing after the second test (something I've been experiencing in `master` branch, as well). @JillieBeanSim Could you please double-check that the integration tests look good on your end? If so, I will be happy to approve this PR. Apologies for the inconvenience, and thanks in advance!

Thanks @fswarbrick for implementing this additional capability!

**Comment by JillieBeanSim** [comment]
@lauren-li I just re-ran unit and integration tests with the merges from master and all look good, I only get the one failing test which is also on master branch.

**Comment by lauren-li** [comment]
> @lauren-li I just re-ran unit and integration tests with the merges from master and all look good, I only get the one failing integration test which is also on master branch.

Great, thank you @JillieBeanSim! I think this PR is good to go in now.

**Review by lauren-li — APPROVED** [review]

**Pull request merged**